### ISSUE TITLE
Host DNS broken after installing calico-vpp

### ISSUE THREAD

**Issue by lwr20** [projectcalico/vpp-dataplane#138]
**Environment**
- Calico/VPP version: 0.13.0
- Kubernetes version: 1.20
- Deployment type: AWS VMs, tried Ubuntu 18.04 and Ubuntu 20.04, installed using kubeadm.
- Network configuration: Used https://raw.githubusercontent.com/projectcalico/vpp-dataplane/release/\
v0.13.0-calicov3.17.1/yaml/generated/calico-vpp-nohuge.yaml .  Set VPP interface to ens5 and pod-cidr to 192.168.0.0/16.

**Issue description**
After applying calico-vpp-nohuge.yaml, the nodes were unable to do DNS lookups, so core-dns and calico-kube-controllers pods were stuck in ImagePullBackoff.

**To Reproduce**
- Setup cluster
- kubectl apply -f calico-vpp-nohuge.yaml

**Expected behavior**
I'd expect core-dns and calico-kube-controllers pods to be pulled correctly.

**Additional context**
Discussed with Aloys in slack.  

Running `sudo tcpdump -v -n -i vpptap0 port 53` I don't see anything when I run `nslookup www.google.com` on one of my hosts.  The nslookup gets
```
** server can't find www.google.com: SERVFAIL
```
instantly (no delay or timeout)

This works though:
```
ubuntu@ip-172-16-101-168:~$ nslookup microsoft.com 172.16.0.2
Server:		172.16.0.2
Address:	172.16.0.2#53
Non-authoritative answer:
Name:	microsoft.com
Address: 40.112.72.205
Name:	microsoft.com
Address: 40.113.200.201
Name:	microsoft.com
Address: 13.77.161.179
Name:	microsoft.com
Address: 104.215.148.63
Name:	microsoft.com
Address: 40.76.4.15
```
> Aloys Augustin  2 days ago
Could you try running :
`sudo systemd-resolve --interface vpptap0 --set-dns 172.16.0.2`
If that solves the issue, I’ll look into how we can do it automatically when configuring the interface

And that did indeed solve the issue.

**Comment by AloysAugustin** [comment]
Thanks @lwr20 for opening this. In order to fix this without writing distro-specific code, the approach we have taken is to rename the vpptap0 interface to the name of the original interface. This has a number of consequences because the initial interface linux configuration will be applied to the tap interface provided by VPP. As a result, we'll also have to switch the interface type to tap instead of tun, to support deployments using DHCP (such as in EC2).

**Comment by AloysAugustin** [comment]
Hi @lwr20 ,

We recently merged some changes that should fix this issue. I just pushed two images with the `prerelease` tag (for the `node`and `vpp` images) that you should be able to try. Let us know if you are able to try them and if they fix the problem :)

**Comment by AloysAugustin** [comment]
Btw - these images should fix #139 as well.

**Comment by lwr20** [comment]
I'll give them a spin tomorrow.

**Comment by AloysAugustin** [comment]
This should be fixed in the 0.14 release, please reopen if it is not the case.
